src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -560,6 +560,14 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
 
             if (hasErrors)
             {
+                if (GetTupleCardinality(left).HasValue &&
+                    GetTupleCardinality(right).HasValue &&
+                    (kind == BinaryOperatorKind.Equal || kind == BinaryOperatorKind.NotEqual))
```

Comment:
what about ```<``` and ```>``` :)

Should we maybe 'lift' all operators onto tuples, just like we can with nullable?  You could then have:

```c#
var x = GetVector()
var y = GetOtherVector();
var z = x + y;
```

While it expands the original proposal, it seems like it would be fine.  And since you're here anyways... :)

AFAICT, it would be fine.  An operator is 'lifted' up to two tuples, if they have the same length, and if the operator is defined for all pairwise matching element types of the tuples.  

What do you think? (Honestly, not trying to feature creep.  It just seems like it would make sense and would fit in well here).